iosApp/iosAppTests/Pages/StopDetails/StopDetailsFilteredViewTests.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  StopDetailsFilteredViewTests.swift
+//  iosAppTests
+//
+//  Created by Kayla Brady on 7/2/25.
+//  Copyright © 2025 MBTA. All rights reserved.
+//
+
+@testable import iosApp
+
+import Combine
+import Shared
+import SwiftUI
+import ViewInspector
+import XCTest
+
+final class StopDetailsFilteredViewTests: XCTestCase {
+    override func setUp() {
+        executionTimeAllowance = 60
+    }
+
+    @MainActor func testSaveEnhancedFavoriteTriggersSaveFlow() throws {
+        let objects = ObjectCollectionBuilder()
+        let stop = objects.stop { _ in }
+        let route = objects.route()
+        let directionId: Int32 = 0
+
+        let favoritesRepository = MockFavoritesRepository()
+        let pinnedRoutesRepository = MockPinnedRoutesRepository()
+
+        let stopDetailsVM = StopDetailsViewModel(
+            favoritesRepository: favoritesRepository,
+            pinnedRoutesRepository: pinnedRoutesRepository
+        )
+        let sut = StopDetailsFilteredView(stopId: stop.id,
+                                          stopFilter: .init(routeId: route.id, directionId: directionId),
+                                          tripFilter: nil,
+                                          setStopFilter: { _ in },
+                                          setTripFilter: { _ in },
+                                          routeCardData: [],
+                                          now: Date.now,
+                                          errorBannerVM: .init(),
+                                          nearbyVM: .init(),
+                                          mapVM: .init(),
+                                          stopDetailsVM: stopDetailsVM)
+
+        let tappedPublisher = PassthroughSubject<Void, Never>()
+
+        let tapButtonExp = sut.inspection.inspect(after: 0.5) { view in
+            try view.find(PinButton.self).find(ViewType.Button.self)
+                .tap()
+            tappedPublisher.send()
+        }
+        let confirmationDialogExp = sut.inspection.inspect(onReceive: tappedPublisher, after: 1) { view in
+            XCTAssertTrue(try view.actualView().inSaveFavoritesFlow)
```

Comment:
Not thrilled that testing for the internal `isSaveFavoritesFlow` value was the best I could manage - would rather test for the presence of the FavoriteConfirmationDialog view, but I think this is the limitation of working with state & ViewInspector.